src/renderer/services/neo.js

```diff
@@ -1063,6 +1240,131 @@ export default {
+        const rawTransaction = {
+          from: currentWallet.ethAddress,
+          gasPrice: web3.utils.toHex(2 * 1e9),
+          gasLimit: web3.utils.toHex(210000),
+          to: toAddress,
+          value: _amount,
+          data: web3.utils.toHex('Tx using Aphelion'),
+          nonce: web3.utils.toHex(nonce1),
+        };
+
+        const estimatedGas = await web3.eth.estimateGas(rawTransaction);
+
+        const amountWithFees = amount.plus(toBigNumber((2 / 1e9) * estimatedGas));
+
+        if (amountWithFees > bal) {
+          return reject(`Insufficient ETH for fees! Need ${amountWithFees} but only found ${bal}`);
+        }
+
+        const transaction = new EthereumTx(rawTransaction);
+        transaction.sign(privateKey);
+
+        const hash1 = await web3.eth.sendSignedTransaction(`0x${transaction.serialize().toString('hex')}`);
+        alerts.success(`Transaction Hash: ${hash1.transactionHash} Successfully Sent, waiting for confirmation.`);
+        return resolve(hash1);
+      } catch (e) {
+        console.log(e);
+        return reject(e);
+      }
+    });
+  },
+
+  sendFundsETHToken(toAddress, amount, symbol, contractAddress, abi, decimals) {
+    const currentWallet = wallets.getCurrentWallet();
+    const web3 = new Web3(new Web3.providers.HttpProvider(network.getSelectedNetwork().infuraApi));
+
+    return new Promise(async (resolve, reject) => {
+      try {
+        try {
+          const cAddress = web3.utils.toChecksumAddress(toAddress);
+          console.log(cAddress);
+        } catch (e) {
+          return reject(e);
+        }
+
+        const privateKey = Buffer(currentWallet.ethPrivateKey, 'hex');
+
+        const tokenContract = new web3.eth.Contract(JSON.parse(abi), contractAddress);
+        const balanceTokenWei = await tokenContract.methods.balanceOf(currentWallet.ethAddress).call();
+        const balanceToken = toBigNumber(balanceTokenWei).dividedBy(10 ** (decimals));
+
+        if (amount > balanceToken) {
+          return reject(`Insufficient ${symbol}! Need ${amount} but only found ${balanceToken}`);
+        }
+
+        const nonce1 = await web3.eth.getTransactionCount(currentWallet.ethAddress, 'pending');
+        const nonce2 = await web3.eth.getTransactionCount(currentWallet.ethAddress);
+
+        if (nonce1 !== nonce2) {
+          return reject('Please wait for confirmation of pending transaction!');
+        }
+
+        const _amount = web3.utils.toHex(toBigNumber(amount) * (10 ** decimals));
+
+        const rawTransaction = {
+          from: currentWallet.ethAddress,
+          gasPrice: web3.utils.toHex(2 * 1e9),
+          gasLimit: web3.utils.toHex(210000),
+          to: contractAddress,
+          value: '0x0',
+          data: tokenContract.methods.transfer(toAddress, _amount).encodeABI(),
+          nonce: web3.utils.toHex(nonce1),
+        };
+
+        const balEthTemp = await web3.eth.getBalance(currentWallet.ethAddress);
+        const bal = balEthTemp / (10 ** 18);
```

Comment:
will there be rounding problems with small amounts?

Reply:
no

Reply:
I think you are only correct if the token does not use too many decimals. Maybe there aren't many ETH tokens that use too many decimals though so math like this will work for tokens that use a reasonable number of decimals.